### PR TITLE
Breadcrumb: add accessibility label

### PR DESCRIPTION
**Pull request by ellatrix**
## Description

See https://github.com/WordPress/gutenberg/pull/18132#issuecomment-573633017.

The block label doesn't make se much sense on the block wrapper, as it's right next to the block content and we'd like the remove this wrapper eventually. The block label makes a lot more sense in navigation mode.

Example:

```html
<button type="button" class="components-button" aria-label="Paragraph Block. Row 16. Imagine everything that WordPress can do is available to you quickly and in the same place on the interface. No need to figure out HTML tags, classes, or remember complicated shortcode syntax. That’s the spirit behind the inserter—the (+) button you’ll see around the editor—which allows you to browse all available content blocks and add them into your post. Plugins and themes are able to register their own, opening up all sort of possibilities for rich editing and publishing.">Paragraph</button>
```

Screenshot 2020-01-13 at 14 48 43

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR. .

### CODE REVIEW

**Comment by MarcoZehe** [comment]
The approach is good. Also nice that we got rid of that debounced hack again. :-)

File: packages/block-editor/src/components/block-list/breadcrumb.js [hunk 1]
```diff
@@ -21,12 +25,27 @@ import BlockTitle from '../block-title';
...
+		const index = getBlockIndex( clientId, rootClientId );
+		const { name, attributes } = __unstableGetBlockWithoutInnerBlocks( clientId );
+		return { index, name, attributes };
+	}
+
+	const {
+		index,
+		name,
+		attributes,
+	} = useSelect( selector, [ clientId, rootClientId ] );
```
Comment (youknowriad) [inline packages/block-editor/src/components/block-list/breadcrumb.js]: I still don't like this pattern for useSelect personally, it makes it very hard to understand the relationship between the deps and the selector and whether we forgot a dep or not.

**Comment by talldan** [comment]
Thanks for taking this on! I've also addressed some of the other feedback for #18132 that's separate from this in #19664.

I think it'd be better to merge this one first and then I'll rebase #19664. I'll try to review tomorrow.

File: packages/block-editor/src/components/block-list/breadcrumb.js (hunk 1 above)
Comment (ellatrix) [reply packages/block-editor/src/components/block-list/breadcrumb.js]: What would you do instead rename the constants?

**Review by talldan — COMMENTED** [review]
Looks good to me code-wise, but I'll defer to Riad for approval over the comment he made. Thanks again for handling this.

**Comment by ellatrix** [comment]
Let's merge. All feedback has been addressed. We can iterate if small adjustments need to be made.

File: packages/block-editor/src/components/block-list/breadcrumb.js (hunk 1 above)
Comment (talldan) [reply packages/block-editor/src/components/block-list/breadcrumb.js]: This becomes a bit more succinct with the selector version of `getAccessibleBlockLabel` in #19664:

https://github.com/WordPress/gutenberg/pull/19664/files#diff-1f379cb4d47b0fe36bd77d3b6e1f4b60R26-R30

**Pull request merged**